### ISSUE TITLE
linux-gpib wont install on 686tupac

### ISSUE THREAD

**Issue by joefowler** [usnistgov/quantum-sensors-drivers#4]
**[Original report](https://bitbucket.org/nist_microcal/nist_qsp_drivers/issues/4) by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

This shows up in the middle of the make output 
```
echo >&2;							\
echo >&2 "  ERROR: Kernel configuration is invalid.";		\
echo >&2 "         include/generated/autoconf.h or include/config/auto.conf are missing.";\
echo >&2 "         Run 'make oldconfig && make prepare' on kernel src to fix it.";	\
echo >&2 ;			
```

I don't remember having to do any of that before, so I'm not sure it's actually the problem, but it does say error.

There is also this
```
pcuser@686tupac:~/nist_qsp_drivers/gpib/linux-gpib$ make | grep error
scripts/Makefile.build:44: /home/pcuser/nist_qsp_drivers/gpib/linux-gpib/drivers/gpib/lpvo_usb_gpib/Makefile: No such file or directory
make[6]: *** No rule to make target '/home/pcuser/nist_qsp_drivers/gpib/linux-gpib/drivers/gpib/lpvo_usb_gpib/Makefile'.  Stop.
make[5]: *** [/home/pcuser/nist_qsp_drivers/gpib/linux-gpib/drivers/gpib/lpvo_usb_gpib] Error 2
make[4]: *** [_module_/home/pcuser/nist_qsp_drivers/gpib/linux-gpib/drivers/gpib] Error 2
make[3]: [all-local] Error 2 (ignored)
```

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

You have devices, though. Can you try using one to see if the driver works?

```
pcuser@686tupac:~/nist_qsp_drivers/gpib/linux-gpib$ ll /dev/gpib*
crw-rw---- 1 root gpib 160,  0 Apr 28 12:49 /dev/gpib0
crw-rw---- 1 root gpib 160,  1 Apr 28 12:49 /dev/gpib1
crw-rw---- 1 root gpib 160, 10 Apr 28 12:49 /dev/gpib10
crw-rw---- 1 root gpib 160, 11 Apr 28 12:49 /dev/gpib11
crw-rw---- 1 root gpib 160, 12 Apr 28 12:49 /dev/gpib12
crw-rw---- 1 root gpib 160, 13 Apr 28 12:49 /dev/gpib13
crw-rw---- 1 root gpib 160, 14 Apr 28 12:49 /dev/gpib14
crw-rw---- 1 root gpib 160, 15 Apr 28 12:49 /dev/gpib15
crw-rw---- 1 root gpib 160,  2 Apr 28 12:49 /dev/gpib2
crw-rw---- 1 root gpib 160,  3 Apr 28 12:49 /dev/gpib3
crw-rw---- 1 root gpib 160,  4 Apr 28 12:49 /dev/gpib4
crw-rw---- 1 root gpib 160,  5 Apr 28 12:49 /dev/gpib5
crw-rw---- 1 root gpib 160,  6 Apr 28 12:49 /dev/gpib6
crw-rw---- 1 root gpib 160,  7 Apr 28 12:49 /dev/gpib7
crw-rw---- 1 root gpib 160,  8 Apr 28 12:49 /dev/gpib8
crw-rw---- 1 root gpib 160,  9 Apr 28 12:49 /dev/gpib9
```

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

Yeah I've tried. You can do `python ~nist_qsp_readout/adr_gui/adr_gui.py` to check.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Well, that GPIB driver is an interesting case. We decided, some time ago, to encapsulate another person's driver inside our repo, so we'd know that the driver always existed (even if the maintainer took down his/her subversion repository).

But that means we tend not to notice updates to it. Would you consider locating the original source? I just did this and saw that we are 3 years behind. (!!)

Not sure how to proceed. If you check that out (see below), then check whether it works, then I can integrate it into this repo.

```bash
svn co svn://svn.code.sf.net/p/linux-gpib/code/trunk linux-gpib-code
```

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

I tried that, there isn't a makefile or a configure script when I check it out.

```
pcuser@686tupac:~/nist_qsp_drivers/gpib/linux-gpib-code/linux-gpib$ ls
applications  contrib  examples  m4            README.hp82335
AUTHORS       COPYING  include   Makefile.am   test
bootstrap     doc      INSTALL   NEWS          TODO
ChangeLog     drivers  language  README        usb
configure.ac  etc      lib       README.HAMEG  util
```
So I don't know how to proceed.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Fair enough.

You are supposed to run `./bootstrap`.  Unfortunately, that invokes `aclocal -I m4`, which core dumps. I have no idea why automake's tools would do this. Want to reboot?

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

I just told it to reboot.

**Comment by joefowler** [comment]
**Original comment by Galen O'Neil (Bitbucket: [oneilg](https://bitbucket.org/oneilg), GitHub: [oneilg](https://github.com/oneilg)).**

----------------------------------------

The answer was to update from the tarball. It has configure and Makefile. 
Fixed by b4a4247
